backend/coins/btc/maketx/maketx_test.go

```diff
@@ -312,3 +313,40 @@ func (s *newTxSuite) TestNewTxCoinSelection() {
 	// coins: .5, .3, .1, .1, .9, .8, .6. select .5+.3+.1+.1 to get 1BTC, take .9 to cover the fees.
 	s.check(amount, feePerKb, s.buildUTXO(500*mBTC, 300*mBTC, 100*mBTC, 100*mBTC, 90*mBTC, 80*mBTC, 70*mBTC), s.change(90*mBTC-txSizeFiveInputs), noDust, s.selectCoins(0, 1, 2, 3, 4))
 }
+
+func TestShuffleTxInputsAndOutputs(t *testing.T) {
+	// Create transaction inputs and outputs
+	txIns := []*wire.TxIn{
+		wire.NewTxIn(&wire.OutPoint{Hash: [32]byte{0x01}, Index: 0}, nil, nil),
+		wire.NewTxIn(&wire.OutPoint{Hash: [32]byte{0x02}, Index: 0}, nil, nil),
+		wire.NewTxIn(&wire.OutPoint{Hash: [32]byte{0x03}, Index: 0}, nil, nil),
+	}
+	txOuts := []*wire.TxOut{
+		{Value: 1000000, PkScript: []byte{}},
+		{Value: 2000000, PkScript: []byte{}},
+		{Value: 3000000, PkScript: []byte{}},
+	}
+
+	// Create a new transaction and add inputs and outputs
+	tx := wire.NewMsgTx(wire.TxVersion)
+	tx.TxIn = txIns
+	tx.TxOut = txOuts
+	// Shuffle with constant seed
+	testRand := rand.New(rand.NewSource(1000))
+	maketx.ShuffleTxInputsAndOutputs(tx, testRand)
```

Comment:
change the first line from `package maketx_test` to `package maketx`, then you can access the unexported stuff. this way, you can unexport `ShuffleTxInputsAndOutputs()` again (it should not be exported as it is only used internally).